packages/sync/legacy/class.jetpack-sync-users.php

```diff
@@ -8,8 +10,8 @@
 class Jetpack_Sync_Users {
 	static $user_roles = array();
 
-	static function init() {
-		if ( Jetpack::is_active() ) {
+	public function __construct( Connection_Manager $connection ) {
```

Comment:
If the only thing we're using the connection for is to retrieve is_active in the constructor, why not pass that in as a boolean? Or just not load this class at all if the connection isn't active?

Reply:
Because it might be too early to know if we're connected. But ideally, that shouldn't be the case, so yeah, I like the idea of keeping dependencies more specific and avoiding passing entire objects when possible.